### ISSUE TITLE
header logging using wrong size ?

### ISSUE THREAD

**Issue by yteraoka** [h2o/h2o#462]
I tryed log request / response header using access log format.
If header maco set as last item of fomat then unexpected null character appeared.

like below.

```
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36^@
Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)^@
```

## Environment

```
# cat /etc/redhat-release 
CentOS Linux release 7.1.1503 (Core)

# uname -a
Linux blog.cs7cbidcfcloud.internal 3.10.0-229.11.1.el7.x86_64 #1 SMP Thu Aug 6 01:06:18 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

# cmake --version
cmake version 2.8.11
```

### h2o version

```
# git show
commit 618d584fef83a11635de89cf679e58f7767ab830
Merge: 838d855 14800ae
Author: Kazuho Oku <kazuhooku@gmail.com>
Date:   Wed Aug 26 16:32:38 2015 +0900

    Merge pull request #460 from h2o/kazuho/mruby/server-push
    
    expose server-push interface to mruby
```

### Build

```
cmake .
make
make install
```

## How to reproduce

```
  access-log:
    path: /tmp/test.log
    formt: "%{user-agent}i"
```

**Comment by kazuho** [comment]
Thank you for reporting.  I have been able to reproduce the issue (please refer to the PR above to find the details).  Will be fixed soon.

**Comment by yteraoka** [comment]
I tried #463. It worked fine. Thanks.

**Comment by kazuho** [comment]
Thank you for checking! Merged #463 to master.
